packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -54,20 +61,78 @@ import {
   TYPICAL_MULTIPLICITY_TYPE,
   VariableExpression,
   INTERNAL__PropagatedValue,
+  SimpleFunctionExpression,
 } from '@finos/legend-graph';
 import { getMultiplicityDescription } from './shared/QueryBuilderUtils';
-import {
-  type PackageableElementOption,
-  DATE_FORMAT,
-  buildElementOption,
-} from '@finos/legend-application';
-import format from 'date-fns/format/index';
-import { addDays } from 'date-fns';
+import { buildPrimitiveInstanceValue } from '../stores/QueryBuilderOperatorsHelper';
 import {
   genericType_setRawType,
   instanceValue_changeValue,
   instanceValue_changeValues,
 } from '../stores/QueryBuilderValueSpecificationModifierHelper';
+import {
+  type CustomDateOption,
+  QUERY_BUILDER_DATE_DURATION_UNIT_LABEL,
+  generateDateFunctionExpression,
+  QUERY_BUILDER_CUSTOM_DATE_START_DATE_OPTION,
+  QUERY_BUILDER_DATE_ADJUSTMENT,
+  QUERY_BUILDER_DATE_OPTION,
+  generateCustomAjustDateOption,
+  generateDateOption,
+  reservedCustomDates,
+  generateDateOptionForStartDayOfDateOption,
+  QUERY_BUILDER_DATE_DURATION_UNIT,
+  generateDateAdjustFunctionFromDateOption,
+  QUERY_BUILDER_DAY_OF_WEEK,
+} from '../stores/QueryBuilderDateTimeHelper';
+
+type TypeCheckOption = {
+  expectedType: Type;
+  // The match flag represents if a strict type match is forced or not.
+  // If match is true, it means that options in the date-capability-dropdown
+  // that are not of the same type or of child types will be filtered out or not.
+  match?: boolean;
+};
+
+type CustomDateInputOption = {
+  label: string;
+  value: string;
+};
+
+const QueryBuilderDateOptionsInfoTooltip: React.FC<{
+  children: React.ReactElement;
+  placement?: TooltipPlacement | undefined;
+}> = (props) => {
+  const { children, placement } = props;
+  return (
+    <Tooltip
+      arrow={true}
+      {...(placement !== undefined ? { placement } : {})}
+      classes={{
+        tooltip: 'query-builder__tooltip',
+        arrow: 'query-builder__tooltip__arrow',
+        tooltipPlacementRight: 'query-builder__tooltip--right',
+      }}
+      TransitionProps={{
+        // disable transition
+        // NOTE: somehow, this is the only workaround we have, if for example
+        // we set `appear = true`, the tooltip will jump out of position
+        timeout: 0,
+      }}
+      title={
+        <div className="query-builder__tooltip__content">
+          <div className="query-builder__tooltip__item">
+            <div className="query-builder__tooltip__item__value">
+              More date options will be available by clicking the left button
```

Comment:
I think it's better to make this an `edit` button, you can click on to trigger the popover, also, when you hover on it, you give users hint. We can do this later